src/ot/test/fuzzer.js

```diff
@@ -87,6 +93,25 @@ const generateRandomInsertTextOp = (snapshot) => {
     return op;
 }
 
+// add_mark: ['path', 'offset', 'length', 'mark', 'data'],
+const generateRandomAddMarkOp = (snapshot) => {
+  console.log('generate random add mark operation');
+  const randomLeaf = getRandomLeafWithPath(snapshot.toJSON().document);
+  const randomPath = randomLeaf.path;
+
+  const offset = fuzzer.randomInt(randomLeaf.text.length);
+  const op = Operation.create({
+      object: 'operation',
+      type: 'add_mark',
+      path: randomPath.slice(0, randomPath.length - 1),
```

Comment:
Just curious. Why did my code work before with
`path: randomPath.slice(1);`

Reply:
I think it was just because it was not deep enough to cause any issues? Really not sure how it worked before since it broke as soon as I added the marks in